open_event/api/custom_fields.py

```diff
@@ -0,0 +1,92 @@
+import re
+from flask.ext.restplus.fields import Raw
+import colour
+
+
+EMAIL_REGEX = re.compile(r'\S+@\S+\.\S+')
+URI_REGEX = re.compile(r'(http|https|ftp)://\S+\.\S+')
+
+
+class CustomField(Raw):
+    """
+    Custom Field base class with validate feature
+    """
+    def format(self, value):
+        """
+        format the text in database for output
+        works only for GET requests
+        """
+        if not self.validate(value):
+            print 'Validation of field with value \"%s\" (%s) failed' % (
+                value, str(self.__class__.__name__))
+            # raise MarshallingError
+            # disabling for development purposes as the server crashes when
+            # exception is raised. can be enabled when the project is mature
+        if self.__schema_type__ == 'string':
+            return unicode(value)
+        else:
+            return value
+
+    def validate(self, value):
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://www.codacy.com/app/triplez/open-event-orga-server/file/2663229950/issues/source?bid=3233864&fileBranchId=3330981#l30)